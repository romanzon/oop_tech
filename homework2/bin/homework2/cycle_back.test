# Cyclic graph with back edges

# Create three nodes - a,b,c
CreateNode a 10
CreateNode b 1
CreateNode c 1 
CreateNode d 2 

# Create graph g and add the nodes
CreateGraph g
AddNode g a
AddNode g b
AddNode g c
AddNode g d

# Create edge - a->b->c->a. b-->d-->b.
AddEdge g a b
AddEdge g b c
AddEdge g c a
AddEdge g b d
AddEdge g d b
AddEdge g d c

# Print nodes
ListNodes g

# Print children
ListChildren g a
ListChildren g b
ListChildren g c
ListChildren g d

# Dfs Algorithm
DfsAlgorithm g a c
DfsAlgorithm g a d
DfsAlgorithm g a

FindPath g a -> c d
