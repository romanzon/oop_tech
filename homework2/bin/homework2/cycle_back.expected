# Cyclic graph with back edges

# Create three nodes - a,b,c
created node a with cost 10
created node b with cost 1
created node c with cost 1
created node d with cost 2

# Create graph g and add the nodes
created graph g
added node a to g
added node b to g
added node c to g
added node d to g

# Create edge - a->b->c->a. b-->d-->b.
added edge from a to b in g
added edge from b to c in g
added edge from c to a in g
added edge from b to d in g
added edge from d to b in g
added edge from d to c in g

# Print nodes
g contains: b c d a

# Print children
the children of a in g are: b
the children of b in g are: d c
the children of c in g are: a
the children of d in g are: c b

# Dfs Algorithm
dfs algorithm output g a -> c: a b d c
dfs algorithm output g a -> d: a b d
dfs algorithm output g a: a b d c

found path in g: a b d with cost 4
